feat: Generate Mod Releases

Umpriel opened the pull request:
Following the idea proposed by [beethy](https://www.nexusmods.com/kingdomcomedeliverance2/users/264835)
Added a new feature to create a new Mods-dev folder at the game's root which will contain new releases by mod devs. After the mod developer makes changes to their mod they can press 'R' and the script will generate a ready to ship zip file uniquely named with a timestamp suffix. 

test build: [KCDAutoPacker/feat: release gen test](https://github.com/Umpriel/KCDAutoPacker/releases/tag/1.0)

Umpriel committed:
feat: Generate Mod Releases

Albeoris commented on KCDAutoPacker/Program.cs:
```diff
@@ -70,11 +69,161 @@ static void MainInternal()
+            string? unpackedDir = null;
+            string modDirName = Path.GetFileName(modDir);
+
+            // Find any .unpacked directory in the mod's Data folder and only select the one that matches the mod's folder name
```
This logic seems wrong. The mod is not the contents of the `.unpacked` folder, but the entire contents of the `Mods/<Name>` folder, including a possible `Localization` folder and the `mod.manifest` file.

It seems we should search for all mods, and filter out those that do not contain a ".unpacked" folder.

Umpriel replied:
You can skip reading the following because I pushed new changes that address your second change request and changes the below implementation.

### For Reference only
The current implementation actually handles this correctly.

1. At lines 124-125, we get the mod directory name:
string? unpackedDir = null;
string modDirName = Path.GetFileName(modDir);

2. We use the `.unpacked` directory only as a marker to identify which mods should be included in the release (lines 127-138). This is intentional - we only want to create releases for mods that are under active development (indicated by having an `.unpacked` folder). < we are not copying anything yet. Only creating structure under Release folder.

3. When it comes to actually copying the mod content, we copy EVERYTHING from the mod directory. This happens in `CopyModDirectoryExcludingUnpacked` (lines 201-225), which:
   - Copies all files and directories from the mod folder
   - Only excludes the `.unpacked` folder itself
   - Preserves the entire mod structure including Localization folders, mod.manifest, and any other content

You can see this is executed only after the structure is ready. at line 185:
This copies the entire mod directory (`modDir`), not just the contents of the `.unpacked` folder. The `.unpacked` check is purely used as a filter for which mods to process, not for determining what content gets copied.

I hope this is clear. I made sure that every step is commented, so if you follow the logic, it will become clear.

Albeoris commented on KCDAutoPacker/Program.cs:
```diff
@@ -70,11 +69,161 @@ static void MainInternal()
+                Directory.CreateDirectory(modTempDir); // This will be the root folder inside the zip file
+
+                // Copy the contents of the mod folder into the root folder inside the temp directory
+                CopyModDirectoryExcludingUnpacked(modDir, modTempDir);
```
Why do we need to copy files again?
You can create a new `ZipArchive` and add all the files to it as `Entries`. The code for this is already in SyncFolder, just put it in a separate method and reuse it:

```csharp
        var diskFiles = new Dictionary<String, FileInfo>(StringComparer.OrdinalIgnoreCase);
        foreach (var file in files)
        {
            String relPath = Path.GetRelativePath(unpackedFolder, file);
            diskFiles[relPath] = new FileInfo(file);
        }
        if (!File.Exists(archivePath))
        {
            Console.WriteLine($"Packing mod: {folderName}");
            using (ZipArchive zip = ZipFile.Open(archivePath, ZipArchiveMode.Create))
            {
                foreach (var kv in diskFiles)
                {
                    zip.CreateEntryFromFile(kv.Value.FullName, kv.Key, CompressionLevel.Optimal);
                    Console.WriteLine($"\t Added: {kv.Key}");
                }
            }
            Console.WriteLine($"Done syncing mod: {folderName}");
        }
```

Umpriel replied:
Done!

Albeoris commented on KCDAutoPacker/Program.cs:
```diff
@@ -70,11 +69,161 @@ static void MainInternal()
+            if (Console.KeyAvailable)
+            {
+                var key = Console.ReadKey(intercept: true).Key;
+                if (key == ConsoleKey.R)  // If 'R' is pressed
```
Please add confirmation to avoid accidental presses.
(Like "Presse ENTER to confirm or any other key to cancel")

Umpriel replied:
Done!

Umpriel committed:
PR feedback: Improved release creation efficiency and safety

- Replaced temporary directory copying with direct zip creation
- Added confirmation prompt for release action
- Removed unused CopyModDirectoryExcludingUnpacked method
- Improved code efficiency by eliminating unnecessary I/O operations
